src/book/03-guides/04-general/04-visualmesh.mdx

```diff
@@ -0,0 +1,161 @@
+
+```bash
+cd ~/code/VisualMesh
+mkdir dataset
+./training/make_dataset.py ~/Downloads/repeated_data ./dataset
+```
+
+## Configure the training
+
+The visual mesh looks for a `config.yaml` file in the output folder to set the training parameters. An example file `example_net.yaml` can be found in the root of the repo. We'll create an output folder and copy the example parameters into it:
+
+```bash
+cd ~/code/VisualMesh
+mkdir output
+cp example_net.yaml ./output/config.yaml
+```
+
+With that, we're almost ready to train the network. To save time, change the configuration to run less training epochs (500 epochs can easily take 30 hours to complete on CPU). Open the config file in a text editor, (e.g. `nano output/config.yaml`) and change the line that says `epochs: 500` to say `epochs: 5`. Also note that the paths for the training, testing and validation tfrecords are set in the config file.
+
+## Train the network
+
+The command to train the network is as below. It's a long command, but it's straightforward if we break it down into the individual parts:
+
+```bash
+cd ~/code/VisualMesh
+docker run --gpus all -u $(id -u):$(id -g) -it --rm --volume $(pwd):/workspace visualmesh:latest ./mesh.py train output
+```
+
+- `docker` is the call to docker
+- `run` is the main command being sent to docker
+- `--gpus all` tells docker to use the gpus from your system (remove `--gpus all` if you're running on a system without a GPU)
+- `-it` tells docker to run in interactive mode with a terminal
+- `--rm` tells docker to automatically clean up the temporary files associated with this container
+- `--volume $(pwd):/workspace` tells docker to mount the current working directory as `/workspace` inside the container
+- `visualmesh:latest` tells docker which image we want to run (note that we built this image and gave it this name in an earlier step)
+- `./mesh.py train <path/to/output>` is the command to run inside the container - note that the filesystem inside the container is different to the filesystem outside of the container, but that due to the `--volume` flag, the current working directory is mounted at `/workspace`.
+
+## Test the network
+
+Testing the network is a very similar command to training, we just replace "train" with "test":
+
+```bash
+cd ~/code/VisualMesh
+docker run --gpus all -u $(id -u):$(id -g) -it --rm --volume $(pwd):/workspace visualmesh:latest ./mesh.py test output
+```
+
+Once the test has finished running, you can inspect the output in `output/test/metrics`.
+
+## Using Tensorboard to inspect the process
+
+Tensorboard [https://www.tensorflow.org/tensorboard] is a tool that provides a dashboard for visualising of the training process that runs in the browser. For tensorflow, we just pass the output director to tensorboard:
```

Comment:
Was this meant to be a markdown `[]()` link?

Reply:
```suggestion
(Tensorboard)[https://www.tensorflow.org/tensorboard] is a tool that provides a dashboard for visualising of the training process that runs in the browser. For tensorflow, we just pass the output director to tensorboard:
```
I originally intended it to be a plain link, but I think it fits the overall style of NUbook better for this to be a markdown link.